src/core/radio/radio_platform.cpp

```diff
@@ -115,6 +116,13 @@ extern "C" void otPlatRadioEnergyScanDone(otInstance *aInstance, int8_t aEnergyS
     return;
 }
 
+bool otPlatRadioIsRxStatePolling(otInstance *aInstance)
+{
+    Instance &instance = *static_cast<Instance *>(aInstance);
+
+    return instance.Get<Mac::Mac>().IsRxStatePolling();
+}
```

Comment:
I am not sure, but I think we may need to exclude this under `OPENTHREAD_RADIO` or RCP build (since `Mac::Mac` won't be available under RCP).

Reply:
Ah that's why there were build failures, the polling state should be available to all builds though. If implementing it within the radio_platform layer is a limitation for RCP builds, could we keep the API in the OTLink layer instead?

Reply:
Won't help to move it to `otLink`. RCP cannot see that either. `Mac::Mac` itself is not present on `RCP` build. It is running on host side.